### PR TITLE
fix: inconsistencies between network versions

### PR DESCRIPTION
**Pull request by alexbarnsley**
## Proposed changes

This PR fixes a few things:
- Broadcasting when there's no peers
- Finding peers on v1 networks
- Dealing with v1 endpoints on v2 networks (possibly added as a result of API headers being different - E.g. Phantom's vendor header for API version)
- Getting peers from a peer that uses https
- Transaction timestamp based on current network's Epoch, not Ark's

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] Refactoring (improve a current implementation without adding a new feature or fixing a bug)

## Checklist

- [x] I have read the [CONTRIBUTING](https://docs.ark.io/guidebook/contribution-guidelines/contributing.html) documentation
- [x] Lint and unit tests pass locally with my changes

### CODE REVIEW

**Comment by alexbarnsley** [comment]
:crazypuppy: thanks darling

File: src/renderer/store/modules/peer.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ export default {
      */
     best: (_, getters) => (ignoreCurrent = true, networkId = null) => {
       const peers = getters['bestPeers'](undefined, ignoreCurrent)
-      if (!peers || !peers.length) {
```
Comment (dated) [inline src/renderer/store/modules/peer.js]: why was this removed? 🤔 i guess it still works as intended, but the modified condition does not stop the execution as intended.
